doc/040_backup.rst

```diff
@@ -416,6 +416,12 @@ environment variables. The following lists these environment variables:
     RESTIC_CACHE_DIR                    Location of the cache directory
     RESTIC_PROGRESS_FPS                 Frames per second by which the progress bar is updated
 
+    HOME                                Backup location for caches
+    LocalAppData                        Backup location for caches
+    PATH                                Lookup path for external programs
+    TMPDIR                              Location for temporary files
+    XDG_CACHE_HOME                      Backup location for caches
```

Comment:
I think that this is out of scope for this documentation. These are well established standard variables/paths, that doesn't belong in a list of environment variables that is more specific to restic's workings. Also, most of them are already mentioned/described in the caching section. I suggest we don't include this change, what do you think @fd0 and @MichaelEischer?

Reply:
Knowing where data goes is crucial when operating a security-focused program. The fact that these are all customary doesn't imply that restic uses them, rather than using a custom RESTIC_TEMPDIR, or /dev/shm, or /run/$uid/restic, or anonymous files in its working directory.

As a practical example, if your homedir is encrypted but the rest of the disk is not, it's important to know that and to know exactly how to keep restic from writing outside the homedir. (I think restic only writes encrypted data to TMPDIR, but I'm not 100% sure.)

Finally, restic has actually used different combinations of variables for locating the cache in previous releases, so the variables aren't that well-established...

Reply:
I'm certainly not convinced. Please consider the following:

- The variables that are relevant for caches are mentioned in the caching section, and shouldn't be in here because that is redundant and these types of variables are not the goal of this list. If there's a need to describe that e.g. $HOME is a "backup location" for caches, then that should be described in the caching section which already talks about this type of information.

- $PATH is just too common to justify mentioning here. If we're going to mention such common variables, there are others to mention as well, and it's not a good idea. Same for $TMPDIR which is a common one too.

Reply:
Listing 3 (or actually 4) variables related to the cache location will probably confuse more that it helps. Regarding the `PATH` variable I think that it is that common that the documentation should only mention when it is _not_ used (e.g. if restic had hardcoded absolute paths).

`TMPDIR` on the other hand is useful to know as that way a user can e.g. direct it to a ramdisk.

Reply:
I removed everything except TMPDIR.